### DISCUSSION TITLE
Android - black screen after update from  v1.0.0 to latest v2.x

### DISCUSSION THREAD

**Discussion by crazyhappygame** [axmolengine/axmol#1428] [Q&A] [answered]
I upgraded from 
```4753270d77283be3f3635cc41126d9fb768e7272 (HEAD, tag: v1.0.0)```
to
```
commit fcf3651cdbc9d613f27c53fe48263499384f44a2 (HEAD)
Author: CHP <crazyhappygame@gmail.com>
Date:   Wed Nov 1 16:35:39 2023 +0100
```
After update on my android games I see only black screen.
I checked logs and found following errors:
```
11-08 21:50:47.477 28482 28573 D libEGL  : ERROR: 1:1: '' : syntax error
11-08 21:50:47.477 28482 28573 D axmol debug info: axmol:ERROR: Failed to compile shader, detail: ERROR: 1:1: '' : syntax error
11-08 21:50:47.477 28482 28573 D axmol debug info: 
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: Program object expected.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: A program must be bound.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: Program object expected.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: A program must be bound.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: Program object expected.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: A program must be bound.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: Program object expected.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: A program must be bound.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: Program object expected.
11-08 21:50:47.477 28482 28573 D libEGL  : ANGLE Info:Debug.cpp:183 (insertMessage): GL error: HIGH: A program must be bound.
```
Could you help me what is wrong?

**Comment by halx99** [comment]
seems load shader file failed

**Reply by crazyhappygame** [reply]
How to fix/debug it. I do not have custom shaders

**Reply by rh101** [reply, marked as answer]
Did you compare the `cpp-template-default` project with your own to see what changes are required? For example, to `CMakeLists.txt`, `build.gradle` etc.?

Double check your `proj.android/app/build.gradle` against the one in the `cpp-template-default`. In particular, ensure you have these changes:
https://github.com/axmolengine/axmol/commit/14a715a8f0171756f2bf4a2120def2b6b49bb58c

**Reply by rh101** [reply]
Just out of curiosity, did you try to create a new project and run it to see if it works on your Android device? For example, using this:
`axmol new -p org.axmol.hellocpp -d . -l cpp HelloCpp`

If that works, then you can be certain that it is something you've missed when updating your app, so just compare that new project with yours.

Also, this error is in the code, so you can put a breakpoint on it and trace it back from there:
`08 21:50:47.477 28482 28573 D axmol debug info: axmol:ERROR: Failed to compile shader, detail: ERROR: 1:1: '' : syntax error`

Search for "Failed to compile shader" in the code and you'll see where it is.

**Reply by rh101** [reply]
The thread with more info that may help you update from Axmol 1.0 to 2.0 is: https://github.com/axmolengine/axmol/issues/1279

It seems like we should have that info in it's own section somewhere, because it's a little more involved if there are custom shaders that need to updated too.

**Comment by crazyhappygame** [comment]
@rh101 Thank you. [That ](https://github.com/axmolengine/axmol/discussions/1428#discussioncomment-7519082) solved my problem.

BTW is it possible to minimize set of changes in project files? 
Would it be possible to put this kind of changes in some common code? 
For example method called from CMakeLists.txt
```
ax_setup_app_config(${APP_NAME})
```
or 
```
axmol\core\platform\android\libaxmol\build.gradle
```
